tests/integration/assert_stack.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"strconv"
+	"strings"
+)
+
+type assertStack struct {
```

Comment:
question: Why does this exist?

thought: It looks like a notable complication to our already complicated (and somewhat fragile yet really really important) assertion logic, and it appears that it is only used for a couple of log lines.

suggestion: Remove this complication (if I'm right and it is just used for a log line).

todo: If keeping, please document it very well.

Reply:
the time I spent on implementing it already payed off in double. If you have a large result to assert on (like a commit query) it show exactly where the mismatch happened.
For example, it shows `path: commits[2].links[1].cid` instead of previous `doc: 1`
So, I would want to keep it.
I will add some documentation.

Reply:
Okay nice :)  Thanks Islam.